### PR TITLE
Update dependencies, plugins and fix bugs found by spotbugs

### PR DESCRIPTION
**Pull request by ZIRAKrezovic**

### CODE REVIEW

**Comment by ZIRAKrezovic** [comment]
Hi @cheels @omcal @dzeromski-hazelcast is it possible to get this one merged? Thanks!

File: pom.xml [hunk 1]
```diff
@@ -30,33 +30,33 @@
...
-        <jetty.version>6.1.26</jetty.version>
-        <javax.inject.version>1</javax.inject.version>
-        <guava.version>31.1-jre</guava.version>
-        <xstream.version>1.4.19</xstream.version>
-        <jackson.version>2.13.4</jackson.version>
-        <jackson.databind.version>2.13.4</jackson.databind.version>
-        <httpclient.version>4.5.13</httpclient.version>
-        <mockito.version>5.1.1</mockito.version>
+        <log4j.version>2.21.1</log4j.version>
+        <guava.version>32.1.3-jre</guava.version>
```
Comment (mkraszew) [inline pom.xml:44]: How about version 33.0.0-jre?

File: pom.xml [hunk 2]
```diff
@@ -30,33 +30,33 @@
...
+        <xstream.version>1.4.20</xstream.version>
+        <jackson.version>2.15.3</jackson.version>
```
Comment (mkraszew) [inline pom.xml:46]: 2.16.1

File: pom.xml [hunk 3]
```diff
@@ -30,33 +30,33 @@
...
+        <jackson.databind.version>2.15.3</jackson.databind.version>
```
Comment (mkraszew) [inline pom.xml:47]: 2.16.1

**Review by mkraszew — APPROVED** [review]

**Comment by mkraszew** [comment]
Looks like nobody cares ;-(

**Pull request closed without merging**